### PR TITLE
Added higher precission for microseconds

### LINKED ISSUE #88: Add Sub-millisecond timing precision
## Requirement - what kind of business use case are you trying to solve?

Accurately timing spans at sub-millisecond resolution.

## Problem - what in Jaeger blocks you from solving the requirement?

DateTimeExtensions.ToUnixTimeMicroseconds() first computes integer milliseconds, then scales to microseconds. This unnecessarily removes sub-millisecond precision from all microsecond measurement.

## Proposal - what do you suggest to solve the problem or improve the existing situation?

Update DateTimeExtensions.ToUnixTimeMicroseconds to the following:

```C#
        private const long TicksPerMicrosecond = TimeSpan.TicksPerMillisecond / 1000; 
        private const long UnixEpochMicroseconds = UnixEpochTicks / TicksPerMicrosecond;

public static long ToUnixTimeMicroseconds(this DateTime utcTimestamp)
        {
            // Truncate sub-millisecond precision before offsetting by the Unix Epoch to avoid
            // the last digit being off by one for dates that result in negative Unix times
            long microseconds = utcTimestamp.Ticks / TicksPerMicrosecond;
            return microseconds - UnixEpochMicroseconds;
        }
```

## Any open questions to address

No known open questions.
There is just one test ```JaegerThriftSpanConverterTests.TestConvertSpan()``` that assumes the implementation is Milliseconds * 1000 which implies that the loss of precision is a requirement to be tested for which I'm assuming is not actually the case.

### PR DESCRIPTION
**Pull request by Falco20019**
Fixes #88

### CODE REVIEW

File: src/Jaeger/Util/DateTimeExtensions.cs [hunk 1]
```diff
@@ -20,17 +20,17 @@ public static class DateTimeExtensions
...
-            // Truncate sub-millisecond precision before offsetting by the Unix Epoch to avoid
+            // Truncate sub-microsecond precision before offsetting by the Unix Epoch to avoid
             // the last digit being off by one for dates that result in negative Unix times
-            long milliseconds = utcTimestamp.Ticks / TimeSpan.TicksPerMillisecond;
-            return milliseconds - UnixEpochMilliseconds;
+            long microseconds = utcTimestamp.Ticks / TicksPerMicrosecond;
+            return microseconds - UnixEpochMicroseconds;
         }
+
+        public static long ToUnixTimeMilliseconds(this DateTime utcTimestamp) => utcTimestamp.ToUnixTimeMicroseconds() / 1000;
```
[the remark this answers is not in the document]
Comment (Falco20019) [reply src/Jaeger/Util/DateTimeExtensions.cs]: Only used in Unit Tests right now

[the remark this answers is not in the document]
Comment (rwkarg) [reply src/Jaeger/Util/DateTimeExtensions.cs]: The only place ```To...Milliseconds``` is used is internally for rate limiting which is a ms. based threshold.

The ```JaegerThriftSpanConverter``` was previously (and is still) using ```To...Microseconds``` to render the span. This PR just removes the artificial truncation of precision in that call's implementation.

**Comment by Falco20019** [comment]
@cwe1ss Friendly ping :) Don't know if you have time for reviews right now. But as of now, I can't to any merges with less then one reviewer with write access.

**Comment by Falco20019** [comment]
@cwe1ss I also just saw that the test case was wrong. Past was in the future (since we did subtract -60 days) and Future was in the past (subtracting 60 days)... Fixed that together with using constants for the timestamp. Will merge it if no one has a complaint the next 24 hours.

**Pull request merged**